src/pages/home/report/ReportActionCompose.js

```diff
@@ -512,17 +513,13 @@ class ReportActionCompose extends React.Component {
      * @param {String} emoji
      */
     addEmojiToTextBox(emoji) {
-        const emojiWithSpace = `${emoji} `;
-        const newComment = this.comment.slice(0, this.state.selection.start)
-            + emojiWithSpace
-            + this.comment.slice(this.state.selection.end, this.comment.length);
         this.setState(prevState => ({
             selection: {
-                start: prevState.selection.start + emojiWithSpace.length,
-                end: prevState.selection.start + emojiWithSpace.length,
+                start: prevState.selection.start + emoji.length,
+                end: prevState.selection.start + emoji.length,
             },
         }));
-        this.updateComment(newComment);
+        this.updateComment(ComposerUtils.insertText(this.comment, this.state.selection, emoji));
```

Comment:
Hi, coming from https://github.com/Expensify/App/issues/18515. This PR raise regression _**Emoji gets added in between the text when selecting text and adding new emoji by search and entering**_.

The `this.state.selection` we pass should be the state before it gets updated, it's due to the [state update may be asynchronous](https://legacy.reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous).